plugins/catalog-backend-module-incremental-ingestion/src/database/IncrementalIngestionDatabaseManager.ts

```diff
@@ -546,12 +560,28 @@ export class IncrementalIngestionDatabaseManager {
    * @param entities - DeferredEntity[]
    */
   async createMarkEntities(markId: string, entities: DeferredEntity[]) {
+    const refs = entities.map(e => stringifyEntityRef(e.entity));
+
     await this.client.transaction(async tx => {
+      const existingRefs = new Set(
+        ...(
```

Comment:
This actually won't work. The `Set` constructor expects just one argument that needs to be iterable. What happens if you spread this array into many arguments, is that it will find the first argument (first element of the array) which is a string - an iterable - and make a set of all of the individual characters of that string.

<img width="186" alt="Screenshot 2022-12-03 at 16 04 24" src="https://user-images.githubusercontent.com/3097461/205447534-f6d6e303-d2dc-47a7-85fb-59a5aac3ba35.png">

Reply:
Yeah, I saw that in my local tests after the fact. I'm not sure how the spread operator got there, it's unneeded since the Knex method returns an iterable. I'm working on some updates, and will fix this shortly.

Reply:
Fixed.